test/testSetup.ts

```diff
@@ -99,11 +102,16 @@ export class TestSetup {
     await expect(uncaughtErrors.length).toBe(0);
   }
 
-  public async setUpTestingEnvironment(): Promise<void> {
+  public async setUpTestingEnvironment(repo?: utilities.repoKeywords): Promise<void> {
     utilities.log('');
     utilities.log(`${this.testSuiteSuffixName} - Starting setUpTestingEnvironment()...`);
 
-    this.tempFolderPath = path.join(__dirname, '..', 'e2e-temp');
+    if (repo) {
+      const repoPath = path.join(__dirname, '..', repo);
+      await gitClone(utilities.projectMaps.get(repo) as string, repoPath); // clone the project
+      Env.getInstance().useExistingProject = repoPath; // set the existing project file path
```

Comment:
Not sure changing the value inside env settings is the right thing to do.

We run our tests one at a time, so there are no side effects here, but thinking into the future when customers might be using this test framework where they might be running a number of tests at a time, modifying the env setting might a negative effect on a test expecting a separate existing project.